Riot/Modules/Settings/SettingsViewController.m

```diff
@@ -1129,126 +1472,12 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
-
-        userSettingsEmailStartIndex = 3;
-        userSettingsNewEmailIndex = userSettingsEmailStartIndex + account.linkedEmails.count;
-        userSettingsPhoneStartIndex = userSettingsNewEmailIndex + 1;
-        userSettingsNewPhoneIndex = userSettingsPhoneStartIndex + account.linkedPhoneNumbers.count;
-        userSettingsThreePidsInformation = userSettingsNewPhoneIndex + 1;
-        
-        count = userSettingsThreePidsInformation + 1;
-    }
-    else if (section == SETTINGS_SECTION_NOTIFICATIONS_SETTINGS_INDEX)
-    {
-        count = NOTIFICATION_SETTINGS_COUNT;
-    }
-    else if (section == SETTINGS_SECTION_CALLS_INDEX)
-    {
-        count = CALLS_COUNT;
-
-        if (!RiotSettings.shared.stunServerFallback)
-        {
-            count -= 2;
-        }
-    }
-    else if (section == SETTINGS_SECTION_DISCOVERY_INDEX)
-    {
-        count = self.settingsDiscoveryTableViewSection.numberOfRows;
-    }
-    else if (section == SETTINGS_SECTION_IDENTITY_SERVER_INDEX)
-    {
-        count = IDENTITY_SERVER_COUNT;
-    }
-    else if (section == SETTINGS_SECTION_INTEGRATIONS_INDEX)
-    {
-        count = INTEGRATIONS_COUNT;
-    }
-    else if (section == SETTINGS_SECTION_USER_INTERFACE_INDEX)
-    {
-        count = USER_INTERFACE_COUNT;
-    }
-    else if (section == SETTINGS_SECTION_IGNORED_USERS_INDEX)
-    {
-        if ([AppDelegate theDelegate].mxSessions.count > 0)
-        {
-            MXSession* session = [AppDelegate theDelegate].mxSessions[0];
-            count = session.ignoredUsers.count;
-        }
-        else
-        {
-            count = 0;
-        }
-    }
-    else if (section == SETTINGS_SECTION_CONTACTS_INDEX)
-    {
-        localContactsSyncIndex = count++;
-        
-        if ([MXKAppSettings standardAppSettings].syncLocalContacts)
-        {
-            localContactsPhoneBookCountryIndex = count++;
-        }
-        else
-        {
-            localContactsPhoneBookCountryIndex = -1;
-        }
-    }
-    else if (section == SETTINGS_SECTION_ADVANCED_INDEX)
-    {
-        count = 1;
-    }
-    else if (section == SETTINGS_SECTION_OTHER_INDEX)
-    {
-        count = OTHER_COUNT;
-    }
-    else if (section == SETTINGS_SECTION_LABS_INDEX)
-    {
-        count = LABS_COUNT;
-    }
-    else if (section == SETTINGS_SECTION_FLAIR_INDEX)
-    {
-        // Check whether some joined groups are available
-        if ([groupsDataSource numberOfSectionsInTableView:tableView])
-        {
-            if (groupsDataSource.joinedGroupsSection != -1)
-            {
-                count = [groupsDataSource tableView:tableView numberOfRowsInSection:groupsDataSource.joinedGroupsSection];
-            }
-        }
-    }
-    else if (section == SETTINGS_SECTION_DEACTIVATE_ACCOUNT_INDEX)
-    {
-        count = 1;
-    }
-    else if (section == SETTINGS_SECTION_SECURITY_INDEX)
-    {
-        count = SECURITY_COUNT;
-    }
-
-    return count;
+    return _sections[section].rows.count;
```

Comment:
We should add some sanity checks and logs issues.
We had crash around that in the past.

Reply:
👍